ten-simple-rules-dockerfiles.Rmd

```diff
@@ -562,70 +561,64 @@ RUN pip install \
+# 7. Mount datasets at run time {-}
 \rulelabel{7}{rule:mount}
 
-The role of containers is to provide the computing environment, not to encapsulate datasets or custom scripts.
-It is better to include data files and custom scripts from the local machine in the container at runtime, and use the container image primarily for the software and dependencies.
+The role of containers is to provide the computing environment, not to encapsulate (potentially very large) datasets.
+It is better to insert data files from the local machine into the container at runtime, and use the image primarily for the software and dependencies.
 This insertion is achieved by using [_bind mounts_](https://docs.docker.com/storage/bind-mounts/).
 Mounting these files is preferable to using the `ADD`/`COPY` instructions in the `Dockerfile`, because files persist when the container instance or image is removed from your system, and the files are more accessible when the workspace is published.
 If you want to add local files to the container, we prefer `COPY` because it is explicit and you do not need `ADD`'s extra features, which are (a) URL as a source, which can be used instead of a `RUN wget` or `RUN curl` to download files and (b) a local [tar file](https://en.wikipedia.org/wiki/Tar_(computing)) as a source, which you should avoid because such an archive file will be harder for future readers of your `Dockerfile` to make sense of and cannot be properly version controlled.
 Unless specific features are needed, bind mounts are also preferable to [storage volumes](https://docs.docker.com/storage/volumes/).
 
-Standalone _script files_ are distinct from other software, as they are treated as files and managed software, i.e., installed with a package manager and versioned.
-If you developed extensive software for a specific analysis, you should ideally publish it as a software package and follow \ruleref{rule:pinning} for installing it (cf. "A package first" in @gruening_recommendations_2019).
-Consider using a suitable package system with pinned versions even for small scripts if the functions are reusable across datasets or workflows by you or others.
-If you cannot publish a software project in the appropriate language's package structure, you should make sure the source code files you `COPY` into the image are published together with the `Dockerfile` (see&nbsp;\ruleref{rule:publish}).
-If the code and `Dockerfile` are in the same repository, the origin and results of the `COPY` instruction and subsequent installation code will be understandable in the future.
-
-Storing _data files_ outside of the container allows for handling of very large datasets and datasets with data worthy of protection, e.g., proprietary data or private information.
-Do not include such data in an image.
+Storing _data files_ outside of the container allows handling of very large or sensitive datasets, e.g., proprietary data or private information.
+Do not include such data in an image!
```

Comment:
Such large data?

Reply:
I think it makes more sense to leave it at the more general "Do not include such data in an image!" since we are also talking about sensitive (but not necessarily large) datasets in the previous sentence.

Reply:
I'm :+1: on "such data".